src/broker/cas_execute.c

```diff
@@ -2804,6 +2809,38 @@ ux_execute_array (T_SRV_HANDLE * srv_handle, int argc, void **argv, T_NET_BUF *
 	{
 	  break;
 	}
+      continue;
+
+    exec_retry_xasl_error:
+      if (session != NULL)
+	{
+	  db_close_session (session);
+	  session = NULL;
+	}
+
+      session = db_open_buffer (srv_handle->sql_stmt);
+      if (!session)
+	{
+	  goto exec_db_error;
+	}
+
+      if (srv_handle->prepare_flag & CCI_PREPARE_XASL_CACHE_PINNED)
+	{
+	  db_session_set_xasl_cache_pinned (session, true, false);
+	}
+
+      stmt_id = db_compile_statement (session);
+      if (stmt_id < 0)
+	{
+	  goto exec_db_error;
+	}
+
+      retried_query_num = num_query;
+
+      srv_handle->session = session;
+      srv_handle->q_result->stmt_id = stmt_id;
+
+      num_query--;
```

Comment:
What about rewrite codes from `exec_retry_xasl_error:` to the end as a function for readability.

Reply:
I made part of exec_retry_xasl_error: into a function.